src/main/java/com/ruralsourcing/WeatherBot/service/HttpCalls.java

```
@@ -27,10 +27,10 @@
      * @throws IOException when http request fails
      */
     public static String request() throws IOException{
-        CloseableHttpClient httpClient = HttpClients.createDefault();
+        CloseableHttpClient httpClient = HttpClients.createDefault(); //The HTTP client that does the HTTP stuff
 
         try{
```

`CloseableHttpClient` implements `Closeable`, so you can use `try-with-resources` here to clean up your code a bit.

Instead of creating the `httpClient` outside of the `try` block and calling `httpClient.close()` in the `finally` block, that can be re-written as:

```java
try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
    // all your other code already in the try-block
} catch (final IOException e) {
    // Also changed this
    e.printStackTrace();
    return null;
}
```

But I would also remove the `catch` block: The `request` method `throws IOException`, so you should just let it propagate to the caller instead of catching it. So that then just becomes:

```java
try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
    // all your other code already in the try-block
}
```